src/language-html/ast.js

```diff
@@ -7,6 +7,21 @@ const NODES_KEYS = {
 
 class Node {
   constructor(props = {}) {
+    /**
+     * define some class properties here, to work around
+     * https://github.com/microsoft/TypeScript/issues/26811
+     */
+    this.index = undefined;
+    this.siblings = undefined;
+    this.prev = undefined;
+    this.next = undefined;
+    this.parent = undefined;
+
+    this.name = undefined;
+    this.namespace = undefined;
+    this.hasExplicitNamespace = undefined;
+    this.children = undefined;
```

Comment:
This looks so silly...

Reply:
I did put the rationale in the comments. I would like to enable JSDoc type checking without excessive `@ts-ignore` comments and without too much overhaul at this point. Any better ideas?

Reply:
I have no idea.

Reply:
The constructor seems taking something like `import('angular-html-parser/lib/compiler/src/ml_parser/ast').Node`, but not exactly the same, I don't know how to make it work.